atomate/vasp/database.py

```diff
@@ -243,47 +208,112 @@ def insert_gridfs(self, d, collection="fs", compress=True, oid=None, task_id=Non
         oid = oid or ObjectId()
         compression_type = None
 
+        # always perform the string conversion when inserting directly to gridfs
+        d = json.dumps(d, cls=MontyEncoder)
         if compress:
             d = zlib.compress(d.encode(), compress)
             compression_type = "zlib"
 
         fs = gridfs.GridFS(self.db, collection)
+        m_data = {"compression": compression_type}
         if task_id:
-            # Putting task id in the metadata subdocument as per mongo specs:
-            # https://github.com/mongodb/specifications/blob/master/source/gridfs/gridfs-spec.rst#terms
-            fs_id = fs.put(
-                d,
-                _id=oid,
-                metadata={"task_id": task_id, "compression": compression_type},
-            )
-        else:
-            fs_id = fs.put(d, _id=oid, metadata={"compression": compression_type})
+            m_data["task_id"] = task_id
+        # Putting task id in the metadata subdocument as per mongo specs:
+        # https://github.com/mongodb/specifications/blob/master/source/gridfs/gridfs-spec.rst#terms
+        fs_id = fs.put(d, _id=oid, metadata=m_data)
 
         return fs_id, compression_type
 
-    def get_band_structure(self, task_id):
+    def insert_maggma_store(self, data: Any, collection: str, oid: ObjectId=None, task_id: Any=None):
+        """
+        Insert the given document into a Maggma store, first check if the store is already
+
+        Args:
+            data: the document to be stored
+            collection (string): the name prefix for the maggma store
+            compress (bool): Whether to compress the data or not
+            oid (ObjectId()): the _id of the file; if specified, it must not already exist in GridFS
+            task_id(int or str): the task_id to store into the gridfs metadata
+        Returns:
+            file id, the type of compression used.
+        """
+        oid = oid or str(ObjectId())
+        compression_type = None
+
+        # make sure the store is availible
+        if collection not in self.maggma_stores:
```

Comment:
As we discussed, you can just call `self.get_obj_store(collection)` here without the if statement.

I would add a check that that the returned store is not None. If it is None, raise an error that the maggma store could not be created. I would also intelligently check the value of self._maggma_store_type, and if that is None, raise the point that no maggma store has been initialised.